src/data.rs

```diff
@@ -27,20 +27,23 @@ impl Deref for HandlePath {
         &self.0
     }
 }
+impl Into<HandleId> for &HandlePath {
```

Comment:
Small nit: I'd prefer `From` here ([relevant lint](https://rust-lang.github.io/rust-clippy/master/#from_over_into)), unless you have a reason for using `Into` instead.

I don't think it matters too much, but just for consistency.

Reply:
I briefly thought this would be an issue with orphan rules but it still works.